### ISSUE TITLE
Experience on my machine with almost exact same specs

### ISSUE THREAD

**Issue by mfischl** [merken/Inspiron7586#2]
## Experience report
...the only difference is I have 16 gigs of ram.

### Things that work
- 4k/Screen/Graphics: Flawless, just perfect as far as I can tell
- Touch screen: Flawless, even with gestures
- Trackpad: Flawless with gestures
- Keyboard: I don't know what layout you have, but on German one I cannot get any keys that require `Alt-Gr` to work yet. But that might be solvable with the correct keyboard layout settings in plain macOS.
- Speaker sound: Works but sounds poorly
- Bluetooth: Works and is discoverable, only tried connecting Sennheiser 4.40BT but that worked flawlessly. Also reconnecting on its own worked perfectly. Only microphone on it didn't work.
- Second screen with USB C: Flawless
- Camera: Flawless
- Internal microphone: Flawless 
- Battery info: Flawless

### Things that don't work
- Airdrop: Does not work, not discoverable by other original macbook and vice versa
- Microphone on bluetooth headset: Does not work, just silent
- Same black screen issue after boot, though I think everything is initialized since on install on the same issue I got visually impaired help to automatically talk to me
- Airplay: Does not discover Samsung TV, did not try any other device, but Samsung works with original macbook
- FN keys: Only volume and play/skip buttons work, brightness and others don't
- Hard clicking triggers same action as three finger swipe up
  - Hard clicking on applications/icons opens their preview
  - That seems to be the lookup feature, can be disabled and makes it a little better, but still doesn't allow dragging dock icons for example
- USB hot plug does not work
- SD card hot plug does not work
- Second screen with hdmi crashes and reboots on plugging in
- Wifi doesn't automatically reconnect on boot, even though automatic reconnect is enabled

Man this is awesome though, I am beyond happy! That is like a 300% upgrade from Windows, thank you for your work! I was really lost for making open core stuff work but I hope I can help on runtime issues a little bit now 🙌 
Tell me if I need to watch out for something else.

**Comment by mfischl** [comment]
Some corrections:
- USB hot plugging works for 3.0 devices on one port only, but on that one it does work
- WIFI is reconnecting automatically if added to the list of known networks manually

**Comment by merken** [comment]
Found a post on Tonyx86mac that might fix the the 3 minute black screen, will try tomorrow...

**Comment by mfischl** [comment]
Also, with https://github.com/hacker1024/Dell-Inspiron-7586-Hackintosh/blob/big_sur/EFI/OC/Kexts/USBMap.kext/Contents/Info.plist I got all two USB ports to work with both USB 2.0 and 3.0 devices fully. Maybe you could add that.

**Comment by hacker1024** [comment]
@mfischl Damn, none of my 3.0 ports can be hot-plugged with my own map on my laptop with identical specs. Lucky.

As for your hard-clicking issues, this is actually a new feature of VoodooI2C, intended to simulate the force click feature from real MacBook trackpads. Disable "Force Click and haptic feedback" in the trackpad preferences and you should be good to go.

**Comment by merken** [comment]
Brightness keys work with FN+S and FN+B (Tip: **S**creen **B**rightness) 😃 
I've updated the repo (new kexts, and a possible black screen fix).

Please report your findings

**Comment by mfischl** [comment]
That is amazing news! 🎉 
I will try and report back as soon as I can

**Comment by mfischl** [comment]
I got around to trying this now. 4 out of 4 times it booted directly without a black screen, which is great success 🎉  But, there is a major caveat: Wifi doesn't work anymore at all. It shows in the top bar as greyed out and after booting I can not click it. After some time I can but it just doesn't work. It doesn't show any SSIDs to connect to and also when going to Network settings, the WIFI adapter doesn't even show. I tried putting the old Airport and itlwm kexts instead of the updated ones but that yielded the same result. Did you change any other configuration related to WIFI?
EDIT: I should mention I am still on Catalina. Did you create that configuration now for BigSur?

**Comment by merken** [comment]
@mfischl Yes, I changed it to itlwm instead of Airport, please install and use the Heliport app included in this repo to connect to wifi: https://github.com/merken/Inspiron7586/blob/main/tools/HeliPort.dmg

**Comment by hacker1024** [comment]
Has anyone tried AirportItlwm? I've found it makes Bluetooth quite unstable, with devices disconnecting frequently. Regular itlwm reduces the disconnections, but doesn't completely fix the issue either.

**Comment by mfischl** [comment]
@merken Just wow... This update now made booting super fast, without a single black screen yet, plus sleep is working now!
I close the lid, I open the lid and its there within a split second. Not a single kernel panic or anything yet. Absolutely great work!

Something I noticed is that battery percentage would sometimes jump great lengths. Like, I opened it now with 38% and suddenly it jumped to 6%.

**Comment by mfischl** [comment]
Small correction now: After a reboot sleep is broken again. Subsequent reboots didn't solve the issue. I did reboot before also, but at a certain point it broke. Still not sure  what caused it.

**Comment by merken** [comment]
Updated for Big Sur (new Kexts + config.plist changes)
